apps/mula/lib/components.ex

```diff
@@ -78,4 +78,78 @@ defmodule Mula.Components do
     </div>
     """
   end
+
+  attr(:value, :float,
+    required: true,
+    doc:
+      "the value represented on the meter, must be between min_value and max_value, if min_value and max_value are not passed, value will be a plain percentage"
+  )
+
+  attr(:max_value, :float,
+    default: 100.0,
+    doc:
+      "the maximun value of the interval represented on the meter, must be greater than min_value, must be greater than or equal value"
+  )
+
+  attr(:min_value, :float,
+    default: 0.0,
+    doc:
+      "the minimun value of the interval represented on the meter, must be lesser than max_value, must be lesser than or equal value"
+  )
+
+  attr(:value_text_formatter, :any,
+    default: nil,
+    doc:
+      "a function that receives the calculated percentage and returns a string, it's used to set the aria-valuetext and the value_text returned"
+  )
+
+  slot(:inner_block, doc: "the optional inner block that renders the")
+
+  def meter(assigns) do
+    %{
+      value: value,
+      max_value: max_value,
+      min_value: min_value,
+      value_text_formatter: value_text_formatter
+    } = assigns
+
+    unless max_value > min_value do
+      raise ArgumentError,
+        message: "[Mula.Components.meter/1] max_value must be greater than min_value"
+    end
+
+    unless max_value >= value do
+      raise ArgumentError,
+        message: "[Mula.Components.meter/1] max_value must be greater than or equal to value"
+    end
+
+    unless value >= min_value do
+      raise ArgumentError,
+        message: "[Mula.Components.meter/1] min_value must be lesser than or equal to value"
+    end
+
+    percentage = (value - min_value) / (max_value - min_value) * 100
+
+    value_text =
+      unless is_nil(value_text_formatter),
+        do: apply(value_text_formatter, [percentage]),
+        else: "#{percentage}%"
```

Comment:
I think all of that logic can be another meter function that does all that assigns by pattern matching if percentage is nil or something like that (like input field is done on core_components). This way we can keep it neat by dividing between what is logic from markup.

Reply:
You mean to use guard clauses to compare `max_value`, `min_value` and `value` in other to raise the correct error? I took a look at the input field on core_components and I am not sure how to use the same logic to solve this problem.

Maybe you meaning that we should not raise `ArgumentError` at all?

We can discuss it further sometime to clarify this points.

Reply:
I changed some things, at this commit: https://github.com/samaraoliveiram/mula/pull/12/commits/db6d6d0a1e7f9710a1ff77e3eb6f8c23470d0bf5 with your latest suggestions.